rules/consistent-words-for-concepts/rule.md

```diff
@@ -0,0 +1,69 @@
+### Cognitive Load
+Using multiple terms to describe the same thing adds cognitive load and increases the mental effort required to understand the codebase. Consistency reduces this burden and makes it easier for anyone to follow the logic.
+
+## Scenario
+Imagine you’re working on an e-commerce app, and you’ve been tasked with adding SMS and email notifications when an order is shipped. You notice the word “consignment” used throughout the codebase and assume this is the relevant concept. You begin searching for a `SendConsignment` method but can’t find it. Assuming no notification logic exists, you go ahead and implement a `SendConsignment` method along with the required dependencies and notification options.    
+
+Later, you open a pull request, and a colleague calls, confused, asking why you’ve re-implemented an entire feature. After some back and forth, they show you the existing `SendOrder` method, which already handles notifications.
+
+:::bad
+```csharp
+public void SendOrder(NotificationType type)
+{ 
+    // existing implementation
+}
+```
+:::
+
+## Outcome
+What went wrong here? Was it a lack of diligence in searching for existing functionality, or the use of different terms — “order” and “consignment” — interchangeably? Likely a bit of both, but primarily, the issue is rooted in inconsistent terminology.    
+
+In this case, the terms “order” and “consignment” may seem related, but they have distinct meanings. An order is something a customer places, indicating their intent to purchase goods or services. A consignment, on the other hand, refers to what a supplier sends to a customer, which may partially or fully fulfill an order.    
+
+In this scenario, the `SendOrder` method should have been called `SendConsignment`, assuming “consignment” was already used in the codebase. 
+
+:::good
+```csharp
+public void SendConsignment(NotificationType type)
+{ 
+    // new implementation
+}
+```
+:::
+
+
+To clarify, it's not necessarily wrong to have a `SendOrder` method, if the term order is ubiquitous. It might represent a pipeline for example, tracking a workflow from submission by the customer to receipt by the customer, and everything in between. But if “order” was the chosen term, the team should have used it consistently across the code. Any introduction of new terminology, such as “consignment,” should be a proactive, team-wide decision that includes any necessary refactoring.
+
+## DRY Principle Implications
+In a worst-case scenario, someone unfamiliar with the `SendOrder` method might merge the `SendConsignment` code without realizing it’s redundant. Now, two methods exist for the same function — each handling notifications differently. This violates the DRY principle, as you now have two distinct pieces of knowledge on handling order shipments, potentially leading to divergent behavior and increased maintenance overhead.
```

Comment:
I thought about this. But I think it needs its own rule. And then I thought I'd just add it because if I don't do it know it will get lost in the mists of time.

So, done! 😀